trunk/wc_yotpo.php

```diff
@@ -103,30 +104,80 @@ function wc_yotpo_uninstall() {
 		delete_option('yotpo_settings');	
 	}	
 }
-function wc_yotpo_show_widget() {		 
+// REVIEWS WIDGET
+function wc_yotpo_show_reviews_widget() {		 
 	global $product;
-	if($product->get_reviews_allowed() == true) {	
-		$product_data = wc_yotpo_get_product_data($product);
-		$yotpo_div = "<div class='yotpo yotpo-main-widget'
-	   				data-product-id='".$product_data['id']."'
-	   				data-name='".$product_data['title']."' 
-	   				data-url='".$product_data['url']."' 
-	   				data-image-url='".$product_data['image-url']."' 
-	  				data-description='".$product_data['description']."' 
-	  				data-lang='".$product_data['lang']."'
-                    data-price='".$product->get_price()."'
-                    data-currency='".get_woocommerce_currency()."'></div>";
-		echo $yotpo_div;
+	if($product->get_reviews_allowed() == true) {
+		echo generate_reviews_widget_code();
+	}						
+}
+function generate_reviews_widget_code() {
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_degault_settings());
+	$use_v3_widgets = $yotpo_settings['use_v3_widgets'];
+	$reviews_widget_id = $yotpo_settings['reviews_widget_id'];
```

Comment:
Why do you get the widget ID here? should be part of the `generate_v3_reviews_widget_code` method.